packages/pwa-buildpack/lib/cli/parse-translations.js

```diff
@@ -0,0 +1,180 @@
+
+module.exports.builder = {
+    coreDevMode: {
+        type: 'boolean',
+        desc:
+            'For core @magento/pwa-studio repository development. Creates a .env file populated with examples if one is not present.'
+    }
+};
+
+module.exports.handler = function buildpackCli(
+    { directory }
+) {
+    const projectRoot = resolve(directory);
+    const projectConfig = loadEnvironment(projectRoot);
+    const config = projectConfig.section('MAGENTO_BACKEND');
+    process.env.MAGENTO_BACKEND_URL = config.url;
+
+    async function queryTranslations(locale, phrases) {
+        return await graphql.getTranslations(locale, phrases);
+    }
+
+    async function queryAvailableLocales() {
+        return await graphql.getAvailableLocales();
+    }
+
+    /** Results are our list of files we want to then extract phrases from */
+    const results = [];
+
+    /**
+     * walk() function recursively identifies .js files in defined directory
+     *
+     * @param {string} dir
+     */
+    const walk = dir => {
+        const list = fs.readdirSync(dir);
+        list.forEach(function(file) {
+            file = resolve(dir, file);
+
+            if (checkExcludedDirs(file) === true) {
+                return;
+            }
+
+            const stat = fs.statSync(file);
+            if (stat && stat.isDirectory()) {
+                const newResults = walk(file);
+                results.concat(newResults);
+            } else {
+                if (file.slice(-3) == '.js') {
+                    results.push(file);
+                }
+            }
+        });
+
+        return results;
+    };
+
+    /**
+     * Excluded specific directories from being parsed to save processing time
+     * @param {string} fileName
+     */
+    const checkExcludedDirs = fileName => {
+        let result = false;
+        const excludedDirs = [
+            '__mocks__',
+            '__tests__',
+            '__stories__',
+            '/queries/'
+        ];
+
+        excludedDirs.forEach(function(element) {
+            if (fileName.includes(element)) {
+                result = true;
+            }
+        });
+
+        return result;
+    };
+
+    /**
+     * phrases object is collection of all parsed phrases in all directories
+     */
+    let phrases = {};
+
+    /**
+     * All directories we want to parse phrases for
+     */
+    const directories = [directory + '/src', '../venia-ui/lib'];
+
+    /**
+     * Retrieve phrases for each directory defined
+     * Uses our walk() method from above
+     */
+    directories.forEach(function(element) {
+        prettyLogger.info(`Retrieving phrases from ${element} directory`);
+        walk(element);
+    });
+
+    prettyLogger.info('Extracting phrases from parsed files');
+    const parser = new Parser();
+    results.forEach(function(result) {
+        parser.parseFuncFromString(fs.readFileSync(result), { list: ['_t'] });
+    });
```

Comment:
Brilliant! This is a beautiful feature of i18next used very well. While I appreciate you making a new Buildpack command (really, this is the first one I've seen, and it's great), I wonder if this could be done as a target integration with the Webpack compiler.

Webpack [parser hooks](https://webpack.js.org/api/parser/) let you take advantage of the parse that Webpack already does. The https://github.com/Perlmint/ya-i18next-webpack-plugin uses that feature; I forget if we discussed this plugin already. It would be great to automatically gather (and aggressively cache) these translations as part of a normal build, rather than as a separate command to remember to run.

Reply:
I don't think that plugin has been brought up before as I was more looking at react i18n rather than webpack. But looks good initially.